README.md

```diff
@@ -1,2 +1,331 @@
-# protocol
-Specification of the Farcaster Protocol
+# Farcaster Protocol
+
+
+## Contents
+1. [Introduction](#1-introduction)
+2. [Concepts](#2-concepts)
+    1. [Accounts](#21-accounts)
+    2. [Signed Messages](#22-signed-messages)
+    3. [Applications](#23-applications)
+    4. [Hubs](#24-hubs)
+3. [Identity System](#3-identity-system)
+    1. [Account Contract](#31-account-contract)
+    1. [Namespace Contract](#32-namespace-contract)
+    3. [Recovery](#33-recovery)
+4. [Replication](#4replication)
+    1. [Signers](#41-signers)
+    2. [Casts](#42-casts)
+    3. [Actions](#43-actions)
+    4. [Verifications](#44-verifications)
+    5. [Metadata](#45-metadata)
+    6. [Root Revocations](#46-root-revocations)
+    7. [Sharding](#47-sharding)
+5. [Peering](#5-peering)
+6. [Upgradeability](#6-upgradeability)
+    1. [Minor Upgrades](#61-minor-upgrades)
+    2. [Major Upgrades](#62-major-upgrades)
+7. [Security Considerations](#7-security-considerations)
+    1. [Signer Compromise](#71-signer-compromise)
+    2. [Eclipse Attacks](#72-eclipse-attacks)
+    3. [Flooding Attacks](#73-flooding-attacks)
+    4. [DDOS Attacks](#74-ddos-attacks)
+    5. [Replay Attacks](#75-replay-attacks)
+8. [URIs](#8-uris)
+9. [Governance](#9-governance)
+
+# 1. Introduction
+
+Social networks have become an essential part of our lives over the last decade. Many of them began their journey as open platforms, courting developers to build on their APIs. These developers created new clients, discovered new UI paradigms, and even launched multi-billion dollar businesses that brought in many users. But over the last few years networks have turned away from developers. They've restricted APIs, implemented arbitrary review processes and have removed access with little notice or recourse. 
+
+Farcaster is a [sufficiently decentralized](https://www.varunsrinivasan.com/2022/01/11/sufficient-decentralization-for-social-networks) protocol that empowers developers to build novel social networks. We define a sufficiently decentralized network as one where **two users who want to communicate are always able to, even if the network wants to prevent it**. Users on such networks must have full control over their identity (usernames), data (messages) and social graph (relationships to others). If a third party controls any of these, they can prevent two users from communicating. Developers must also be free to build applications and have unrestricted access to the network and users must be free to switch between them. If there was only one app to connect to the network, it could prevent two users from communicating.  
```

Comment:
This is good, but sounds more like a messaging protocol than a social network. It also sounds like "you can't block people", which isn't quite the point.

Especially given the first paragraph, it might be worth defining it first as a `a network on top of which anyone can build a client, and users are empowered to control where there data is stored as well as own their identity, data, and social graph`.